qiskit/visualization/__init__.py

```diff
@@ -17,20 +17,160 @@
 
 .. currentmodule:: qiskit.visualization
 
+
+Overview
+========
+
+The visualization module contain functions that visualizes measurement outcome counts, quantum
+states, circuits, pulses, devices and more. 
+
+Prerequisites
+=============
+
+To use visualization functions, you are required to install visualization optionals to your
+development environment:
+
+.. code-block:: bash
+
+   pip install 'qiskit[visualization]'
+
 Counts and State Visualizations
 ===============================
 
+This section contains functions that visualize measurement outcome counts and quantum states.
+
+APIs
+----
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
 .. autosummary::
    :toctree: ../stubs/
 
    plot_histogram
+
+State Visualization
+^^^^^^^^^^^^^^^^^^^
+
+.. autosummary::
+   :toctree: ../stubs/
+
    plot_bloch_vector
    plot_bloch_multivector
    plot_state_city
    plot_state_hinton
    plot_state_paulivec
    plot_state_qsphere
 
+Example Usage
+-------------
+
+Counts Visualization
+^^^^^^^^^^^^^^^^^^^^
+
+Here is an example of using :func:`plot_histogram` to visualize measurement outcome counts:
+
+.. jupyter-execute::
+
+   from qiskit.tools.visualization import plot_histogram
+
+   counts = {"00": 501, "11": 499}
+   plot_histogram(counts)
+
+The data can be a dictionary with bit string as key and counts as value, or more commonly a
+:class:`~qiskit.result.Counts` object obtained from :meth:`~qiskit.result.Result.get_counts`.
+
+State Visualization
+^^^^^^^^^^^^^^^^^^^
+
+Here is an example of using :func:`plot_state_city` to visualize a quantum state:
+
+.. jupyter-execute::
+
+   from qiskit.tools.visualization import plot_state_city
+
+   state = [[0+1.0j, 0-1.0j],
+            [1-0.5j, 1+0.5j]]
```

Comment:
Have conflicting feelings about this matrix. On the one hand it shows some variety of values and that's nice. On the other hand this is not a valid density matrix at all.

Reply:
Change to this matrix which is the state of a quantum circuit initialize with 0 and applied RY(pi/3):
```
state = [[ 0.75   , 0.433j],
         [ -0.433j, 0.25  ]]
```